src/test/groovy/com/synacy/gradprogram/spock/exercise/OrderingServiceTest.groovy

```diff
@@ -0,0 +1,199 @@
+        given:
+        List itemListWithEachType = [foodItem, applianceItem, clothingItem, gadgetItem]
+        Cart cartWithEachItemType = new Cart(UUID.randomUUID(), itemListWithEachType)
+
+        when:
+        boolean containsFoodItem = orderingService.cartContainsFoodItem(cartWithEachItemType)
+
+        then:
+        containsFoodItem
+    }
+
+    def "cartContainsFoodItem should return false if Cart has no Item with type FOOD"() {
+        given:
+        List itemListWithoutFood = [applianceItem, clothingItem, gadgetItem]
+        Cart cartWithoutFoodItem = new Cart(UUID.randomUUID(), itemListWithoutFood)
+
+        when:
+        boolean containsFoodItem = orderingService.cartContainsFoodItem(cartWithoutFoodItem)
+
+        then:
+        !containsFoodItem
+    }
+
+    def "calculateTotalCostOfCart should return the sum of the cost of each item in cart"() {
+        given:
+        List itemListWithEachType = [foodItem, applianceItem, clothingItem, gadgetItem]
+        Cart cartWithEachItemType = new Cart(UUID.randomUUID(), itemListWithEachType)
+        double expectedTotalCost = 1710.0
+
+        when:
+        double actualTotalCost = orderingService.calculateTotalCostOfCart(cartWithEachItemType)
+
+        then:
+        expectedTotalCost == actualTotalCost
+    }
+
+    def "isCartEligibleForDiscount should return true if total cost of cart is greater than 50.0 and items in cart is greater than 5"() {
+        given:
+        List itemListForDiscount = [foodItem, applianceItem, clothingItem, gadgetItem, foodItemLowCost, applianceItemLowCost, gadgetItemLowCost]
+        Cart cartForDiscount = new Cart(UUID.randomUUID(), itemListForDiscount)
+
+        when:
+        boolean eligibleForDiscount = orderingService.isCartEligibleForDiscount(cartForDiscount)
+
+        then:
+        eligibleForDiscount
+    }
+
+    def "isCartEligibleForDiscount should return false if total cost of cart is less than 50.0"() {
+        given:
+        List itemListWithLowCostManyItems = [foodItem, foodItemLowCost, clothingItemLowCost, applianceItemLowCost, gadgetItemLowCost, foodItemLowCost, foodItemLowCost]
+        Cart cartWithLowCostManyItems = new Cart(UUID.randomUUID(), itemListWithLowCostManyItems)
+
+        when:
+        boolean eligibleForDiscount = orderingService.isCartEligibleForDiscount(cartWithLowCostManyItems)
+
+        then:
+        !eligibleForDiscount
+    }
+
+    def "isCartEligibleForDiscount should return false if items in cart is less than 5"() {
+        given:
+        List itemListWithHighCostFewItems = [gadgetItem, applianceItem]
+        Cart cartWithHighCostFewItems = new Cart(UUID.randomUUID(), itemListWithHighCostFewItems)
+
+        when:
+        boolean eligibleForDiscount = orderingService.isCartEligibleForDiscount(cartWithHighCostFewItems)
+
+        then:
+        !eligibleForDiscount
+    }
+
+    def "applyDiscountToCartItems should set cost of each item in cart to 10 percent of original"() {
```

Comment:
I think we also need to add a test to check if the cart is not eligible for discount.